lighthouse-core/audits/non-composited-animations.js

```diff
@@ -0,0 +1,178 @@
+const ACTIONABLE_FAILURE_REASONS = [
+  {
+    flag: 1 << 13,
+    text: UIStrings.unsupportedCSS,
+  },
+];
+
+/**
+ * Return list of actionable failure reasons and a boolean if some reasons are not actionable.
+ * @param {number} failureCode
+ * @return {{failureReasons: string[], hasNonActionable: boolean}}
+ */
+function getActionableFailureReasons(failureCode) {
+  /** @type {string[]} */
+  const failureReasons = [];
+  ACTIONABLE_FAILURE_REASONS.forEach(reason => {
+    if (failureCode & reason.flag) {
+      failureReasons.push(reason.text);
+      failureCode &= ~reason.flag;
+    }
+  });
+  return {failureReasons, hasNonActionable: Boolean(failureCode)};
+}
+
+class NonCompositedAnimations extends Audit {
+  /**
+   * @return {LH.Audit.Meta}
+   */
+  static get meta() {
+    return {
+      id: 'non-composited-animations',
+      scoreDisplayMode: Audit.SCORING_MODES.INFORMATIVE,
+      title: str_(UIStrings.title),
+      description: str_(UIStrings.description),
+      requiredArtifacts: ['traces', 'AnimatedElements'],
+    };
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @param {LH.Audit.Context} context
+   * @return {Promise<LH.Audit.Product>}
+   */
+  static async audit(artifacts, context) {
+    const trace = artifacts.traces[Audit.DEFAULT_PASS];
+    const traceOfTab = await TraceOfTab.request(trace, context);
+
+    /** @type {Map<string, {begin: LH.TraceEvent | undefined, status: LH.TraceEvent | undefined}>} */
+    const animationPairs = new Map();
+    traceOfTab.mainThreadEvents.forEach(event => {
+      if (event.name !== 'Animation') return;
+
+      if (!event.id2 || !event.id2.local) return;
+      const local = event.id2.local;
+
+      if (!animationPairs.has(local)) {
+        animationPairs.set(local, {begin: undefined, status: undefined});
+      }
+
+      const pair = animationPairs.get(local);
+      if (!pair) return;
+      switch (event.ph) {
+        case 'b':
+          pair.begin = event;
+          break;
+        case 'n':
+          pair.status = event;
+          break;
+      }
+    });
+
+    /** @type Map<string, {failureReasons: string[], nodes: LH.Audit.Details.NodeValue[]}> */
+    const animations = new Map();
+    animationPairs.forEach(pair => {
+      if (!pair.begin ||
+          !pair.begin.args.data ||
+          !pair.begin.args.data.nodeId ||
+          !pair.status ||
+          !pair.status.args.data ||
+          !pair.status.args.data.compositeFailed) return;
+
+      const nodeId = pair.begin.args.data.nodeId;
+      const element = artifacts.AnimatedElements.find(e => e.nodeId === nodeId);
+      if (!element) return;
+      /** @type LH.Audit.Details.NodeValue */
+      const node = {
+        type: 'node',
+        path: element.devtoolsNodePath,
+        selector: element.selector,
+        nodeLabel: element.nodeLabel,
+        snippet: element.snippet,
+      };
+
+      // Report animation only if all failure reasons are actionable
```

Comment:
might be nice to add the why.

what are some examples of unactionable reasons? is it a temporary situation that could be resolved such that it's still worthwhile to warn the user about the remaining fixable reasons?

Reply:
You make a good point though that actionable reasons are worth reporting if the non actionable reason is temporary.  It might be worth discussion about each failure reason but I will take this out for now.